# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build libvorbis-build sdl2-build"

PKG_CONFIG_PATH=/dfb/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

pkg-config --exists flac-build && REQUIRES="$REQUIRES flac-build"

# configure (ac-2.69; lt-2.2.6)
libtoolize -c -f -i; cat acinclude/* > aclocal.m4; autoconf
PKG_CONFIG_PATH=/dfb/lib/pkgconfig LDFLAGS=-Wl,-rpath,/dfb/lib ./configure --disable-static --prefix=/dfb

# build
make

# install
install -d $DESTDIR/dfb/bin
install build/.libs/playmus $DESTDIR/dfb/bin/playmus2
install build/.libs/playwave $DESTDIR/dfb/bin/playwave2
install -d $DESTDIR/dfb/include/SDL2
install -m 644 SDL_mixer.h $DESTDIR/dfb/include/SDL2
install -d $DESTDIR/dfb/lib
install build/.libs/libSDL2_mixer-2.0.so.0.* $DESTDIR/dfb/lib/libSDL2_mixer-2.0.so.0
ln -sf libSDL2_mixer-2.0.so.0 $DESTDIR/dfb/lib/libSDL2_mixer.so
install -d $DESTDIR/dfb/lib/pkgconfig
install -m 644 SDL2_mixer.pc $DESTDIR/dfb/lib/pkgconfig

# build.pc
install -d $DESTDIR/dfb/share/pkgconfig
cat > $DESTDIR/dfb/share/pkgconfig/sdl2_mixer-build.pc << EOF
Name: SDL2_mixer
Version: 2.0.2
Description: Audio mixer library for SDL
Requires: $REQUIRES

devel=\\
/dfb/include/SDL2/SDL_mixer.h \\
/dfb/lib/libSDL2_mixer.so \\
/dfb/lib/pkgconfig/SDL2_mixer.pc

exec=\\
/dfb/bin/playmus2 \\
/dfb/bin/playwave2 \\
/dfb/lib/libSDL2_mixer-2.0.so.0
EOF

exit
--- SDL2_mixer-2.0.2.orig/configure.in
+++ SDL2_mixer-2.0.2/configure.in
@@ -1,6 +1,7 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_INIT(README.txt)
 AC_CONFIG_AUX_DIR(build-scripts)
+AC_CONFIG_MACRO_DIR([acinclude])
 
 dnl Set various version strings - taken gratefully from the GTk sources
 
@@ -179,7 +180,7 @@
         host_lib_path="/usr/$base_libdir /usr/local/$base_libdir"
     fi
     for path in $env_lib_path $gcc_bin_path $gcc_lib_path $host_lib_path; do
-        lib=[`ls -- $path/$1 2>/dev/null | sed -e '/\.so\..*\./d' -e 's,.*/,,' | sort | tail -1`]
+        lib=[`ls -- $path/$1 2>/dev/null | sed -e '/\.so\..*\./d' -e 's,.*/,,' | sort | tail -n -1`]
         if test x$lib != x; then
             echo $lib
             return
